### ISSUE TITLE
Library backprop interop

### ISSUE THREAD

**Issue by leifu1128** [huggingface/candle#803]
So say I have an ```op``` from a foreign library (that also has their own autodif system), and I can easily call a method to get the gradients of ```op```'s inputs w.r.t to it's output (assume it's scalar for now). How hard would be to support backprop through it's output and what might such an implementation look like? I apologize if this is difficult to answer, it's probably highly dependent on the foreign library's autodif implementation. 

For context, I am looking to train a model to fit volatility surfaces for options. And I would like to use some ops from RustQuant as part of my loss function.

**Comment by LaurentMazare** [comment]
Not sure it would totally fit the bill but I would suggest looking at `CustomOp` if you haven't already done so. It's the way to add custom operations in candle and it supports specifying the backprop step for the newly introduced op.
You can see an [example of this in the tests](https://github.com/huggingface/candle/blob/4f18180fc7564b9a02cab13c4acda0ac7b17a799/candle-core/tests/custom_op_tests.rs#L95-L99). I don't think the backward override has been used much so far so certainly keen to get some feedback based on how it goes for you.
